mongoDB-case-study.md

```diff
@@ -0,0 +1,113 @@
+- 2019 - Released the version 4.2 - MongoDB Atlas Search and MongoDB Atlas Data Lake. And MongoDB Operator for Kubernetes and MongoDB Connector for Apache Kafka.
+- 2020 - Released the version 4.4 - MongoDB Atlas Online Archive, Realm & Sync, MongoDB Atlas multi-cloud clusters.
+- 2021-2022 - Released the version 5.0, and Rapid released - MongoDB 5.x, MongoDB Stable API, Atlas Serverless instances, MongoDB Atlas Data API
+- 2022 - Released the version 6.0 - Atlas Serverless instances, Atlas Data API, Atlas CLI and Flexible Sync. Atlas Data Lase, Atlas Data Federation and Atlas SQL Interface. And Cluster-to-cluster sync.
+
+## MongoDBs main Features
+
+- Indexing
+  - This makes the searching though the database less time consuming and makes it easier. This means that every field in the documents are indexed with primary and secondary indices.
+- Replication
+  - This means the database makes copies of the data and sends it to multiple servers. Which make the data more available and redundant, if one server fails, the data can be retrieved from another server. A replica set can contain of two or more copies of the data, each set may act in the role of primary or secondary replica. Where it is the primary replica that serves as the main server and interacts with the client and performs the read/write operations. The secondary replica is a copy of the data of the primary using built-in replication. If the primary replica should fail, it automatically switches to the secondary, that then serves as the primary server.
+- Aggregation
+  - Makes it possible to perform operations on the grouped data, and then receive a single or computed result. This is thanks to the three different aggregation methods; aggregation pipeline, map-reduce function and a single-purpose aggregation methods.
+- Scalability
+  - Through sharding you distribute the data to different servers, where the data is partitioned into chunks using a shard key. These chunks are evenly distributed across shards thats placed on different servers. This way MongoDB provides horizontal scalability, which balances the load to keep the system up and running, and gives less chance of hardware failure.
+- Document-oriented
+  - Each database contains a collection which contains documents. And each document can be different, with variations in number of fields and size.
+  - The documents structure can be familiar to developers, as they are constructed like objects.
+- Ad hoc queries
+  - Supports search by field, range queries and by regular expression searches.
+
+## Strengths
+
+- It is schema-less - you don't need to design any schema.
+- It's flexible.
+- High-performing
+- Available and Scalable.
+- It replications makes it less prone to hardware failures.
+- Programming languages - they have official drives for multiple programming languages, both major and a large number of smaller once.
+
+## Weaknesses
+
```

Comment:
You should mention ACID here

Reply:
I've fixed the other issues/suggestions, this is a good point and I will need some more time to dig into this!